### PR TITLE
tiup: add TiUP overview doc

### PR DESCRIPTION
**Pull request by yikeke**
### What is changed, added or deleted? (Required)

add TiUP overview doc

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/2878
- Other reference link(s): website link: https://pingcap.com/docs-cn/stable/reference/tools/tiup/overview/

Note: This PR references several other TiUP docs, so it should be merged after those other PRs.

### CODE REVIEW

File: reference/tools/tiup/overview.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
+---
+title: TiUP Overview
+summary:
```
Comment (yikeke) [inline reference/tools/tiup/overview.md:3]:
```suggestion
summary: Introduce the TiUP tool and its ecosystem.
```

File: reference/tools/tiup/overview.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+> **Note:**
+>
+> - The number of available components will continue to grow. To check the latest supported components, execute the `tiup list --refresh` command.
+> - The list of available versions of components will also continue to grow. To check the latest supported component versions, execute the `tiup list <component> --refresh` command.
+
+TiUP commands are implemented in TiUP's internal code and used for package management operations, while TiUP components are independent component packages installed by TiUP commands.
+
+For example, if you run the `tiup list` command, TiUP directly runs its own internal code; if you run the `tiup playground` command, TiUP first checks whether there is a local package named "playground", and if not, TiUP downloads the package from the mirror, and then run it.
+
+All TiUP commands are described in [Managing Components with TiUP](/reference/tools/tiup/manage-component.md). All TiUP components are divided into the following topics by component:
```
Comment (yikeke) [inline reference/tools/tiup/overview.md:133]:
```suggestion
All TiUP commands are described in [Managing Components with TiUP Commands](/reference/tools/tiup/manage-component.md). All TiUP components are divided into the following topics by component:
```

File: reference/tools/tiup/overview.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  $ tiup clean <name>                  # Clean the data of running/terminated instance (Kill process if it's running)
+  $ tiup clean --all                   # Clean the data of all running/terminated instances
+
+Use "tiup [command] --help" for more information about a command.
+```
+
+The output is long but you can focus on only two parts:
+
+- Available commands
+    - install: used to install components
```
[the remark this answers is not in the document]
Comment (yikeke) [reply reference/tools/tiup/overview.md:108]: It's a phrase. I don't think it's necessary?

[the remark this answers is not in the document]
Comment (TomShawn) [reply reference/tools/tiup/overview.md:108]: OK

**Comment by TomShawn** [comment]
LGTM

File: reference/tools/tiup/overview.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+TiUP commands are implemented in TiUP's internal code and used for package management operations, while TiUP components are independent component packages installed by TiUP commands.
+
+For example, if you run the `tiup list` command, TiUP directly runs its own internal code; if you run the `tiup playground` command, TiUP first checks whether there is a local package named "playground", and if not, TiUP downloads the package from the mirror, and then run it.
+
+All TiUP commands are described in [Managing Components with TiUP Commands](/reference/tools/tiup/manage-component.md). All TiUP components are divided into the following topics by component:
+
+- [Quickly Deploy a Local TiDB Cluster](/reference/tools/tiup/playground.md): introduce the playground component
+- [Deploy and Maintain an Online TiDB Cluster](/reference/tools/tiup/cluster.md): introduce the cluster component
+- [Build a Private Mirror](/reference/tools/tiup/mirrors.md): introduce the mirrors component
```
Comment (yikeke) [inline reference/tools/tiup/overview.md:137]:
```suggestion
- [Create a Private Mirror](/reference/tools/tiup/mirrors.md): introduce the mirrors component
```

File: reference/tools/tiup/overview.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+- [Package a TiUP component](/reference/tools/tiup/package-component.md): introduce the package component
```
Comment (yikeke) [inline reference/tools/tiup/overview.md:138]:
```suggestion
- [Package a Component](/reference/tools/tiup/package-component.md): introduce the package component
```

**Pull request merged**